src/test/java/catan/controllers/testcases/play/BuyDevCardTest.java

```diff
@@ -0,0 +1,360 @@
+
+@RunWith(SpringJUnit4ClassRunner.class)
+
+//@SpringApplicationConfiguration(classes = {TestApplicationConfig.class, RequestResponseLogger.class})
+@SpringApplicationConfiguration(classes = TestApplicationConfig.class)
+@WebIntegrationTest("server.port:8091")
+public class BuyDevCardTest extends PlayTestUtil {
+
+    public static final String USER_NAME_1 = "user1_BuyDevCardTest";
+    public static final String USER_PASSWORD_1 = "password1";
+    public static final String USER_NAME_2 = "user2_BuyDevCardTest";
+    public static final String USER_PASSWORD_2 = "password2";
+    public static final String USER_NAME_3 = "user3_BuyDevCardTest";
+    public static final String USER_PASSWORD_3 = "password3";
+
+    private static boolean initialized = false;
+
+    @Autowired
+    private RandomUtil randomUtil;
+
+    private Scenario scenario;
+
+    @Before
+    public void setup() {
+        if (!initialized) {
+            registerUser(USER_NAME_1, USER_PASSWORD_1);
+            registerUser(USER_NAME_2, USER_PASSWORD_2);
+            registerUser(USER_NAME_3, USER_PASSWORD_3);
+            initialized = true;
+        }
+
+        scenario = new Scenario((RandomUtilMock) randomUtil);
+    }
+
+    @Test
+    public void should_fail_when_trying_to_buy_card_before_dice_is_thrown() {
+        playPreparationStage()
+
+                .getGameDetails(1).gameUser(1).hasAvailableAction("THROW_DICE")
+                .getGameDetails(1).gameUser(1).doesntHaveAvailableAction("BUY_CARD")
+
+                .BUY_CARD(1).failsWithError("ERROR");
+    }
+
+    @Test
+    public void should_fail_when_trying_to_buy_card_not_in_players_turn() {
+        playPreparationStage()
+
+                .BUY_CARD(2).failsWithError("ERROR");
+    }
+
+    @Test
+    public void should_successfully_buy_card_after_dice_was_thrown() {
+        playPreparationStage()
+
+                .THROW_DICE(1)
+                .getGameDetails(1).gameUser(1).hasAvailableAction("BUY_CARD")
+
+                .BUY_CARD(1).successfully();
+    }
+
+    @Test
+    public void should_successfully_buy_second_card_after_first_card_was_already_bought_in_current_turn() {
+        playPreparationStage()
+
+                .THROW_DICE(1)
+
+                .BUY_CARD(1).successfully()
+                .BUY_CARD(1).successfully();
+    }
+
+    @Test
+    public void should_return_bought_card_in_response() {
+        playPreparationStage()
+
+                .THROW_DICE(1)
+                .nextRandomDevelopmentCards(asList(KNIGHT, VICTORY_POINT, ROAD_BUILDING, MONOPOLY, YEAR_OF_PLENTY))
+
+                .BUY_CARD(1).boughtCardIs(KNIGHT)
+                .BUY_CARD(1).boughtCardIs(VICTORY_POINT)
+                .BUY_CARD(1).boughtCardIs(ROAD_BUILDING)
+                .BUY_CARD(1).boughtCardIs(MONOPOLY)
+                .BUY_CARD(1).boughtCardIs(YEAR_OF_PLENTY);
+    }
+
+    @Test
+    public void should_increase_development_cards_when_user_buys_a_new_one() {
```

Comment:
it is not the same test case, if I don't return bought card (like it happened today), this test case will pass and 'should_return_bought_card_in_response' will fail. And wise versa, if I return bought card but don't increase user's dev cards, current test case will fail and 'should_return_bought_card_in_response' will pass. 
So I don't think that we should combine test cases that test different things. But we can check everything in a single test case like it was done in PreparationStageTest, but I think that it was very bad idea